Filter styles

rgclayton opened the pull request:
* Initial layout and styling for filtering
* Implements https://github.com/adobe/theblog/issues/193

rofe commented on .model/filters.css:
```diff
@@ -0,0 +1,244 @@
+  border-radius: 0;
+}
+
+.dropdown-menu .options {
```
I think this container could use a bottom padding: the last item ends up sitting right at the bottom border.

rgclayton replied:
That's how it is in the designs.
![image](https://user-images.githubusercontent.com/2539954/84186110-4381b880-aa4d-11ea-9c6b-439718a262a0.png)

rofe replied:
Well, the design doesn't show the end of the list, because there is a scroll bar, but I would assume the last item to have a padding or margin at the bottom...

rgclayton replied:
Makes sense. This is fixed. I've added spacing at the bottom of the dropdown.

rgclayton commented:
Thanks @rofe 

> I think the element alignment below 600px min-width is incorrect: from the design spec it looks like they should be aligned left with a 32px margin so they align with card below.

Ok. I will get this fixed.

> Maybe it would help to integrate the card grid below (merge master for the cards model) so you can adjust the filter bar to it in the different viewports.

Great suggestion. Will do.

> As a bonus, would you like to take a stab at implementing the general mechanics like open/close dropdown, clear selection etc?

Sure, I'll take a stab at this.

rgclayton commented:
Screen grab of the updated alignment on mobile. I'll start working on adding basic JS functionality as discussed.

![image](https://user-images.githubusercontent.com/2539954/84189462-8abe7800-aa52-11ea-9771-768997a9b34e.png)

rgclayton commented:
@rofe I've added JS that will show/hide the drop down menus and clear any filters selected.

rgclayton commented:
> Neat! I think this is ready to be included in the topic page. Probably using an HTML "template" we inject from JS for now. I'll post the list of dropdown items in the underlying GH issue .
> 
> Some comments and open questions from the redlines:
> 
> * When the dropdown menu is open, the page underneath should be dimmed out (white on larger screens, black on mobile)
> * It looks like the grayed out style should stay applied for as long as the dropdown menu is open
> * I'm not exactly sure when that 2px border style should be applied, but maybe it's on hover?
> * On mobile (below 600px), there's a centered, floating affordance with clear all and apply buttons. I assume that's because the idea on larger screens is to keep the dropdown open while (multiple) filter selections will get applied as soon as the list items get clicked - something that can't work on a smaller screen... maybe we should just always have an apply button at the bottom of the dropdown, which closes the dropdown and triggers the search, regardless of the screen resolution?

@rofe  Do want those updates in this PR, or would you prefer a new PR?

rofe commented:
@rgclayton let's keep this one open

rofe commented:
Alternatively, you could also directly start implementing it in `topic.js` in the `staging` branch

rgclayton commented:
@rofe

> When the dropdown menu is open, the page underneath should be dimmed out (white on larger screens, black on mobile)

Added.

> It looks like the grayed out style should stay applied for as long as the dropdown menu is open

Updated

> I'm not exactly sure when that 2px border style should be applied, but maybe it's on hover?

Sounds good to me. Added on hover

> On mobile (below 600px), there's a centered, floating affordance with clear all and apply buttons. I assume that's because the idea on larger screens is to keep the dropdown open while (multiple) filter selections will get applied as soon as the list items get clicked - something that can't work on a smaller screen... maybe we should just always have an apply button at the bottom of the dropdown, which closes the dropdown and triggers the search, regardless of the screen resolution?

Updated dropdown at mobile breakpoint. I added the Apply button to all break points. I thought that was a good suggestion. I added a clear filter next to the apply button for mobile only, since the mobile view covers the clear filters. In the comps for mobile the search area was replaced with a filter title, so i did that. Not sure if that is correct or not? The more I went back and forth between mobile and tablet up the more odd discrepancies I found...

rofe commented:
I like it a lot! I believe the inconsistencies in the mobile view design will sort themselves out once we integrate it into the site and discuss with the biz team. I'll go ahead and merge this now.
